packages/teams-js/src/public/dialog.ts

```diff
@@ -122,13 +133,22 @@ export namespace dialog {
   export function sendMessageToParentFromDialog(
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     message: any,
-  ): void {
+  ): Promise<void> {
     ensureInitialized(FrameContexts.task);
     if (!isSupported()) {
       throw errorNotSupportedOnPlatform;
     }
 
-    sendMessageToParent('messageForParent', [message]);
+    // if actionInfo exists in appContext, the dialog was loaded from a 3rd party app,
+    // and if sendMessageToParentFromDialog was called with a message,
+    // throw an error because the message cannot be passed along to the desired recipient
+    return app.getContext().then((appContext) => {
+      if (message && appContext.actionInfo) {
+        throw errorInvalidArguments;
```

Comment:
If my above comment is true, we would want a different error message here

Reply:
tbh, I don't like this error message at all, but since content actions is new, there is no sdk error that really makes sense.  I would love input on this...adding to the description